features/environment.py

```diff
@@ -134,13 +134,8 @@ def _make_patroni_test_config(self, name, dcs, tags):
 
     @staticmethod
     def _make_connstring(config):
-        connstring = config['postgresql']['connect_address']
-        if ':' in connstring:
-            address, port = connstring.split(':')
-        else:
-            address = connstring
-            port = '5432'
-        return 'host={0} port={1} dbname=postgres user=postgres'.format(address, port)
+        tmp = (config['postgresql']['connect_address'] + ':5432').split(':')
```

Comment:
You are missing [:2] here, otherwise, you will get 3-members tuple if connect address already contains the port, and that will fail on the next line with "tuple index out of range".